app/controllers/api/cloud_object_store_containers_controller.rb

```diff
@@ -1,4 +1,33 @@
 module Api
   class CloudObjectStoreContainersController < BaseController
+    def create_resource(type, _ems_id = nil, data = {})
+      options = {}
+      options[:name] = data["name"] if data["name"]
+
+      # Depending on the storage manager type, collect required form params.
+      case data["emsType"]
+      when "ManageIQ::Providers::Amazon::StorageManager::S3"
+        if data["provider_region"]
+          options[:create_bucket_configuration] = {
+            :location_constraint => data["provider_region"]
+          }
+        end
+      when "ManageIQ::Providers::Openstack::StorageManager::SwiftManager"
+        if data["cloud_tenant_id"]
+          options[:cloud_tenant_id] = data["cloud_tenant_id"]
+        end
+      end
+      create_ems_resource(type, data, :supports => true) do |ems, klass|
+        {:task_id => klass.cloud_object_store_container_create_queue(User.current_userid, ems, options)}
```

Comment:
```suggestion
        {:task_id => klass.cloud_object_store_container_create_queue(User.current_userid, ems, data.symbolize_keys)}
```

I'd feel much more comfortable having the `case` logic put into the individual provider's implementation.

Amazon will need to take `:provider_region` and convert into the `:create_bucket_configuration`
Or alternatively, you can define the form to tell the front end to put the field in the designated place.

OpenStack will be unchanged.